File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

	@Reference
	private BrandService brandService;
	
	@RequestMapping("/findAll")
	public List<TbBrand> findAll() {
		return brandService.findAll();
	}
	
	@RequestMapping("/findBrandByPage")
	public PageResult findBrandByPage(int pageSize, int pageNum) {
		return brandService.findBrandByPage(pageSize, pageNum);
	}
	
	@RequestMapping("/addBrand")
	public Result addBrand(@RequestBody TbBrand brand) {
		try {
			brandService.add(brand);
			return new Result(true, "新增成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "新增失败");
		}
	}
	
	@RequestMapping("/findByID")
	public TbBrand findByID(long id) {
		return brandService.findByID(id);
	}
	
	@RequestMapping("/updateBrand")
	public Result updateBrand(@RequestBody TbBrand brand) {
		
		try {
			brandService.update(brand);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}
	
	@RequestMapping("/deleteByID")
	public Result deleteByID(long[] ids) {
		try {
			brandService.delete(ids);
			return new Result(true, "删除成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
	@RequestMapping("/search")
	public PageResult search(@RequestBody TbBrand brand, int pageSize, int pageNum) {
		return brandService.findPaginResult(brand, pageSize, pageNum);
	}
	
	/**
	 * 查询品牌的下拉列表
	 * @return
	 */
	@RequestMapping("/selectOptionList")
	public List<Map> selectOptionList() {
		return brandService.selectOptionList();
	}
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function($scope, $controller, brandService){
    	
			$controller('baseController',{$scope:$scope});//继承	
			
    		//寻找所有的品牌列表
    		$scope.findAll = function(){
	    		brandService.findAll().success(function(data) {
	    			$scope.list = data;
	    		});
    		}
    		
			
			
			
			    
			//根据分页参数查找品牌列表		
			$scope.findBrandByPage = function(currPage, pageSize) {
				
				brandService.findBrandByPage(currPage, pageSize).success(function(data) {
console.log(data)
					$scope.list = data.rows;
					$scope.paginationConf.totalItems = data.total;
					
				});
			}	
			
			//添加新的品牌信息
			$scope.save = function() {
			
				let method = brandService.addBrand($scope.entity);
				
				if($scope.entity.id != null) {
					method = brandService.updateBrand($scope.entity);
				}
				
				method.success(function(data) {
					if(data.success) {
						$scope.reloadList();
					}else {
						alert(data.message);
					}
				});
			}				
    		
    		//根据id查找品牌信息
    		$scope.findByID = function(id) {
    		
    			brandService.findByID(id).success(function(data) {
    				$scope.entity = data;
    			})
    		}
    		
    		
    		//删除品牌信息
    		$scope.deleteByID = function() {
    		
    			brandService.deleteByID($scope.selectIds).success(function(data) {
    				if(data.success) {
	    				$scope.reloadList();
    				} else {
    					alert(data.message);
    				}
    				
    				//删除不管成功还是失败清空id数组
    				$scope.selectIds = [];
    			})
    		}
    		
    		$scope.searchEntity = {};
    		
    		$scope.search = function(currPage, pageSize) {
    			brandService.search(currPage, pageSize, $scope.searchEntity).success(function(data) {
console.log(data)
					$scope.list = data.rows;
					$scope.paginationConf.totalItems = data.total;
					
				});
    		}
    		
    	});